tools/cpm_tpm_rpk/cpm_tpm_rpk.R

```diff
@@ -1,142 +1,148 @@
   ),
   make_option(
     c("-b", "--gene_header"),
     default = TRUE,
-    type = 'logical',
+    type = "logical",
     help = "Consider first line of gene length as header ? [default : '%default' ]"
   ),
   make_option(
     c("-l", "--log"),
     default = FALSE,
-    type = 'logical',
+    type = "logical",
     help = "Should be log transformed as well ? (log2(data +1)) [default : '%default' ]"
   ),
   make_option(
     c("-o", "--out"),
     default = "res.tab",
-    type = 'character',
+    type = "character",
     help = "Output name [default : '%default' ]"
   ),
   make_option(
     "--visu",
     default = FALSE,
-    type = 'logical',
+    type = "logical",
     help = "performs T-SNE [default : '%default' ]"
   ),
   make_option(
     "--tsne_labels",
     default = FALSE,
-    type = 'logical',
+    type = "logical",
     help = "add labels to t-SNE plot [default : '%default' ]"
   ),
   make_option(
     "--seed",
     default = 42,
-    type = 'integer',
+    type = "integer",
     help = "Seed value for reproducibility [default : '%default' ]"
   ),
   make_option(
     "--perp",
     default = 5.0,
-    type = 'numeric',
+    type = "numeric",
     help = "perplexity [default : '%default' ]"
   ),
   make_option(
     "--theta",
     default = 1.0,
-    type = 'numeric',
+    type = "numeric",
     help = "theta [default : '%default' ]"
   ),
   make_option(
     c("-D", "--tsne_out"),
     default = "tsne.pdf",
-    type = 'character',
+    type = "character",
     help = "T-SNE pdf [default : '%default' ]"
   ),
   make_option(
     "--pca_out",
     default = "pca.pdf",
-    type = 'character',
+    type = "character",
     help = "PCA pdf [default : '%default' ]"
   )
 
 )
 
-opt = parse_args(OptionParser(option_list = option_list),
-                 args = commandArgs(trailingOnly = TRUE))
+opt <- parse_args(OptionParser(option_list = option_list),
+                  args = commandArgs(trailingOnly = TRUE))
 
-if (opt$data == "" & !(opt$help)) {
+if (opt$data == "" && !(opt$help)) {
   stop("At least one argument must be supplied (count data).\n",
        call. = FALSE)
-} else if ((opt$type == "tpm" | opt$type == "rpk") & opt$gene == "") {
+} else if ((opt$type == "tpm" || opt$type == "rpk") && opt$gene == "") {
   stop("At least two arguments must be supplied (count data and gene length file).\n",
        call. = FALSE)
-} else if (opt$type != "tpm" & opt$type != "rpk" & opt$type != "cpm" & opt$type != "none") {
+} else if (opt$type != "tpm" && opt$type != "rpk" && opt$type != "cpm" && opt$type != "none") {
   stop("Wrong transformation requested (--type option) must be : cpm, tpm or rpk.\n",
        call. = FALSE)
 }
 
-if (opt$sep == "tab") {opt$sep = "\t"}
-if (opt$gene_sep == "tab") {opt$gene_sep = "\t"}
+if (opt$sep == "tab") {
+  opt$sep <- "\t"
+}
+if (opt$gene_sep == "tab") {
+  opt$gene_sep <- "\t"
+}
 
 cpm <- function(count) {
-  t(t(count) / colSums(count)) * 1000000
+  (count / colSums(count)) * 1000000
```

Comment:
@bellenger-l  this is the main change with this PR, do you validate ?

Reply:
Yes !